src/auth/ITokenAuthenticator.ts

```diff
@@ -0,0 +1,3 @@
+export interface ITokenAuthenticator {
+    GenerateToken: () => string;
```

Comment:
Suggestion: use a type alias to represent a JWTToken.
```suggestion
    GenerateToken: () => Token;
    
  ...
  type Token = string;
```

Reply:
Good suggestion! I added a JWT type alias for a string. Decided on naming it `JWT` instead of Token since it's less ambiguous about the type of Token.